### ISSUE TITLE
Deleting App process deleted the backend environment of another app

### ISSUE THREAD

**Issue by Lorenzohidalgo** [aws-amplify/amplify-hosting#2891]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### App Id

ddhhh5znoq37b & d3sy22uqr5mvv6

### Region

eu-central-1

### Describe the bug

While deleting APP2 (which did only host the frontend & relayed on the backend of APP1) the process failed and deleted the production environment of APP1.

The backend production environment of APP1 is now deleted and listed under APP2 in the GUI.

### Expected behavior

This should never happen. Not sure even why this has happened. Only the app flagged to be deleted should have been deleted.

### Reproduction steps

1. Create APP1 with backend environment
2. Create APP2 (only frontend hosting relaying on APP1s backend)
3. Build & deploy app2
4. Delete APP2

**Comment by Lorenzohidalgo** [comment]
hey @hloriana just saw that the issue was assigned, I've also opened an AWS Support case, here is the case id in case it's helpfull/needed: 10468207461 
I'm also available in the was amplify discord in case you need any additional details. I'm really hoping that it's possible to restore the deleted environment.

**Comment by hloriana** [comment]
Hi @Lorenzohidalgo 👋🏽 I just walked through the reproduction steps and I was unable to reproduce this behavior. When I deleted App2, the environment from App1 was not deleted. How did you connect the backend from App1 to App2?

**Comment by Lorenzohidalgo** [comment]
@hloriana I didn't connect them explicitly, the hosted front-end app just called all the resources from app1.  Any idea how to recover/restore the deleted environment? It was the production environment for both apps.

**Comment by Lorenzohidalgo** [comment]
@hloriana  Yes, I can find the deleted environment stack there. What should I do?

**Comment by hloriana** [comment]
@Lorenzohidalgo please reach out to aws-amplify-customer@amazon.com with reference to this GitHub issue so we can further assist.

**Comment by Lorenzohidalgo** [comment]
I've just sent the email

**Comment by Lorenzohidalgo** [comment]
Hey @hloriana any update on this?
